source/_components/media_player.kodi.markdown

```diff
@@ -51,3 +52,184 @@ Add music to the default playlist (i.e. playlistid=0).
 | `media_name` | no| Optional media name for filtering media. Can be 'ALL' when `media_type` is 'ALBUM' and `artist_name` is specified, to add all songs from one artist.
 | `artist_name` | no | Optional artist name for filtering media.
 
+
+### {% linkable_title Service `kodi_execute_addon` %}
+
+Run a [Kodi Addon](https://kodi.tv/addons) with optional parameters. Results of the Kodi API call, if any, will be redirected in a Home Assistant event: `kodi_execute_addon_result`.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| `entity_id` | no | Name(s) of the Kodi entities where to run the (pre-installed) Kodi Addon.
+| `addonid` | yes | Name of the Kodi addon.
+| any other parameter | no | Optional parameters for the Kodi API call.
+
+### {% linkable_title Service `kodi_run_method` %}
+
+Run a [Kodi JSONRPC API](http://kodi.wiki/?title=JSON-RPC_API) method with optional parameters. Results of the Kodi API call will be redirected in a Home Assistant event: `kodi_run_method_result`.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| `entity_id` | no | Name(s) of the Kodi entities where to run the API method.
+| `method` | yes | Name of the Kodi JSONRPC API method to be called.
+| any other parameter | no | Optional parameters for the Kodi API call.
+
+
+### {% linkable_title Event triggering %}
+
+When calling the `kodi_run_method` or the `kodi_execute_addon` services, if the Kodi JSONRPC API returns data, when received by Home Assistant it will fire a `kodi_run_method_result` or a `kodi_execute_addon_result` event on the event bus with the following `event_data`:
+
+```yaml
+entity_id: "<Kodi media_player entity_id>"
+result_ok: <boolean>
+input: <input parameters of the service call>
+result: <data received from the Kodi API>
+```
+
+### {% linkable_title Kodi services samples %}
+
+#### Simple script to turn on the TV with the Kodi JSON-CEC Addon
+
+```yaml
+script:
+  activate_tv:
+    alias: Turn on TV
+    sequence:
+      - alias: TV on
+        service: media_player.kodi_execute_addon
+        data:
+            entity_id: media_player.kodi
+            addonid: script.json-cec
+            params:
+                command: activate
+```
+
+#### Dynamic input select with Kodi content
+
+A more complex example, to configure a `input_select` that can play in Kodi the latest entries added when selecting them.
+
+It consists of 3 scripts that query the Kodi API to return the contents of the Kodi library, and an AppDaemon app with which the events are heard with the results of the API queries.
```

Comment:
You mean the complex example, right? Where and how do you think it would be more convenient to place it?

(I've removed the `kodi_execute_addon` service, as discussed in the home-assistant/home-assistant#7603)